lib/mochify.js

```diff
@@ -154,51 +156,60 @@ module.exports = function (_, opts) {
     noColors        : opts.noColors
   };
 
+  (function initPlugins(next) {
+    if (opts.consolify) {
+      b.plugin(consolify, opts);
+      next();
+    } else if (opts.node) {
+      b.plugin(node, opts);
+      next();
```

Comment:
This is one thing I still don't understand: in theory this should be `process.nextTick(next)` as well, so that the execution order of the blocks is always the same:

- initPlugins
- main body + return
- appendAdditionalPlugins

Yet, when I do this, the `node/plugin` tests will start failing as it looks as if the tests all run synchronously before `appendAdditionalPlugins` is even called.

I'm wondering if this is a pointer to a more fundamental problem here?

Reply:
A possible way to solve this could look like this: https://github.com/mantoni/mochify.js/commit/cd4252fe5ef1ac0c47da2a3eae3914b3ed247864

Reply:
I ran a few checks and I presume there seems to be a slightly bigger issue with the `node` runner here. It's adding a through stream to the browserify pipeline to capture the input `js` and when changing the line here to `process.nextTick(next);` that stream ends immediately causing the node plugin to abort.

The way I'm testing these things is by adding `console.error(stderr);` in `test/fixture/run.js` and then put `console.error('!');` statements in the code. I don't have the capacity to fix this right now.

Reply:
We'd be sacrificing quite a bit of performance when delaying the `bundle` call until everything else is running. Browserify will only run the transforms and plugins on the code once `bundle()` is called which is currently happening while chrome starts up, the web server starts up and chrome loads the URL in a tab. The code is currently written to do as much setup concurrently as possible.